internal/server/proto/server.proto

```diff
@@ -1583,6 +1583,9 @@ message ListInstancesRequest {
     // the app are returned.
     Ref.Workspace workspace = 2;
   }
+
+  // Time to wait before retrying a request to connect to requested instance
+  string connect_timeout = 3;
```

Comment:
Nitpick: thinking about this, suggesting this field be named `wait_timeout` instead of connect since we're waiting for instances to appear (which does imply connection so 🤷 )